docs/integrations/argocd-integration.md

```diff
@@ -0,0 +1,411 @@
+``` 
+
+Config-map: 
+
+``` 
+kubectl create configmap ssh-known-hosts --from-file=< path to you known hosts file >
+``` 
+
+ssh config secret
+
+``` 
+kubectl create secret generic ssh-config-secret \
+  --from-file=< path to your ssh config file >
+``` 
+
+Example ssh config file
+
+``` 
+Host github.com
+  HostName github.com
+  IdentityFile ~/.ssh/id_ed25519_github
+``` 
+
+After making changes to the webhook deployment file, apply this yaml in your cluster.
+
+You can also run terrascan admission controller server outside cluster, for more information on configuring terrascan as an admission controller webhook, follow https://docs.accurics.com/projects/accurics-terrascan/en/latest/integrations/admission-controller-webhooks-usage 
+
+
+#### Step 2: Create a Dockerfile for the container which has the terrascan script to run the remote scan against the terrascan’s admission controller webhook.
+
+```DockerFile
+# Dockerfile with a script to use terrascan's validating webhook
+# configured in the kubernetes cluster, to scan a repo for violations
+FROM alpine:3.12.0
+
+#curl to send request to terrascan validating webhook
+RUN apk add --no-cache curl 
+
+WORKDIR /home/terrascan
+
+RUN mkdir bin
+
+COPY scripts/argocd-terrascan-remote-scan.sh  bin/terrascan-remote-scan.sh
+
+# create non root user
+RUN addgroup --gid 101 terrascan && \
+  adduser -S --uid 101 --ingroup terrascan terrascan
+
+RUN chown -R terrascan:terrascan bin && \
+  chmod u+x bin/terrascan-remote-scan.sh
+
+USER terrascan
+
+CMD ["sh"]
+```
+
+terrascan-remote-scan script
+
+```sh
+#!/bin/sh
+TERRASCAN_SERVER="https://${SERVICE_NAME}"
+IAC="k8s"
+IAC_VERSION="v1"
+CLOUD_PROVIDER="all"
+REMOTE_TYPE="git"
+
+SCAN_URL="${TERRASCAN_SERVER}/v1/${IAC}/${IAC_VERSION}/${CLOUD_PROVIDER}/remote/dir/scan"
+
+RESPONSE=$(curl -s -w \\n%{http_code} --location -k  --request POST "$SCAN_URL" \
+--header 'Content-Type: application/json' \
+--data-raw '{
+"remote_type":"'${REMOTE_TYPE}'",
+"remote_url":"'${REMOTE_URL}'"
+}')
+
+echo "$RESPONSE"
+
+HTTP_STATUS=$(printf '%s\n' "$RESPONSE" | tail -n1)
+
+if [ "$HTTP_STATUS" -ne 200 ]; then
+  exit 3
+fi
+```
+
+
+#### Step 3: Configure PreSync hook to use container created in step 2
+
+The following example hook yaml is mostly ready to be added to an existing kubernetes configuration.
+
+```yaml
+apiVersion: batch/v1
+kind: Job
+metadata:
+generateName: terrascan-hook-
+namespace: test
+annotations:
+  argocd.argoproj.io/hook: PreSync            
+spec:
+ttlSecondsAfterFinished: 3600
+template:
+  spec:
+    securityContext:
+      seccompProfile:
+        type: RuntimeDefault
+    containers:
+    - name: terrascan-argocd
+      image: <IMAGE FROM STEP 2>
+      resources:
+        requests:
+          cpu: "1"
+          memory: "256Mi"
+        limits:
+          cpu: "1"
+          memory: "256Mi"
+      env:
+        - name: SERVICE_NAME
+          value: terra-controller-service.default.svc
```

Comment:
this value would be name of the service used to expose admission controller webhook.